service-base/src/main/java/org/eclipse/hono/service/resourcelimits/PrometheusBasedResourceLimitChecks.java

```diff
@@ -678,7 +678,7 @@ long calculateEffectiveLimit(
         Objects.requireNonNull(periodType, "period mode");
 
         if (targetDateTime.isBefore(effectiveSince)) {
-            throw new IllegalArgumentException("target date-time must not be before effective since");
+            return 0;
```

Comment:
As per the master branch, the configured _maxBytes_ value is returned when the _effectiveSince_ date lies in the future. Now by returning zero, it means that no messages are accepted for this tenant (as the `allowedMaxBytes` is zero bytes) until the plan actually starts as per the _effectiveSince_ date. I think it makes more sense not to allow any messages until the plan actually starts.

Reply:
I agree. That's why I made that change ;-)